src/main/java/com/mercateo/spring/security/jwt/security/JWTAuthenticationTokenFilter.java

```diff
@@ -15,50 +15,107 @@
-        String header = request.getHeader(TOKEN_HEADER);
-
-        if (header == null || !header.startsWith("Bearer ")) {
-            final String pathInfo = request.getPathInfo();
-            log.warn("no JWT token found {}{} ({})", request.getServletPath(), pathInfo != null ? pathInfo : "",
-                    header);
-            throw new InvalidTokenException("no token");
+    public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) req;
+        HttpServletResponse response = (HttpServletResponse) res;
+        String tokenHeader = request.getHeader(TOKEN_HEADER);
+
+        if (isInvalidTokenPrefixForBearer(tokenHeader)) {
+            try {
+                handleNoBearerToken(request, response, chain, tokenHeader);
+            } catch (InvalidTokenException e) {
+                unsuccessfulAuthentication(request, response, e);
+            }
         } else {
-            String authToken = header.split("\\s+")[1];
+            super.doFilter(request, response, chain);
+        }
+    }
 
-            return getAuthenticationManager().authenticate(new JWTAuthenticationToken(authToken));
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request,
+            HttpServletResponse response) {
+        String tokenHeader = request.getHeader(TOKEN_HEADER);
+
+        if (isInvalidTokenPrefixForBearer(tokenHeader)) {
+            return null;
         }
+        String authToken = tokenHeader.split("\\s+")[1];
+        return getAuthenticationManager().authenticate(new JWTAuthenticationToken(authToken));
     }
 
     @Override
-    protected void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain,
+    protected void successfulAuthentication(HttpServletRequest request,
+            HttpServletResponse response, FilterChain chain,
             Authentication authResult) throws IOException, ServletException {
         super.successfulAuthentication(request, response, chain, authResult);
 
         chain.doFilter(request, response);
     }
+
+    private boolean isInvalidTokenPrefixForBearer(final String tokenHeader) {
+        return tokenHeader == null || !tokenHeader.startsWith(TOKEN_PREFIX_BEARER);
+    }
+
+    private void handleNoBearerToken(HttpServletRequest request, HttpServletResponse response,
+            FilterChain chain, String token) throws IOException, ServletException {
+        final String pathInfo = String.valueOf(request.getPathInfo()).replace("null", "");
+        final String servletPath = String.valueOf(request.getServletPath()).replace("null", "");
+
+        // request URL depends on the default servlet or mounted location
+        final String pathToCheck = servletPath + pathInfo;
+        log.warn("no JWT token found {} ({})", pathToCheck, token);
```

Comment:
Can this be dropped then? Every "spam" attack will spill up the logs immededately. (Not a new "issue", but perhaps worth a thought.)

Reply:
I'm not sure which log level is right here. I think this depends on several points:
- technologie/mechanisms to protect from spam (should block the (spam) requests before)
- it is desired that requests are performed without a JWT
- how and if you want to monitor that requests are performed without a JWT
- ....

Depending on these points you should configure the loglevel yourself e.g. with logback.xml
```
        <logger name="com.mercateo.spring.security.jwt.security.JWTAuthenticationTokenFilter"
            level="INFO" />
```

Reply:
My point is, why logging a missing token with WARN at all when in the first if-branch it is totally ok, that there is none. Additionally the in the else-case an exception is thrown indicating the issue. (If it is an issue.)

We are currently resorting to adjusting the log level, for not getting these log entries at all, because the client is notified about the issue and requet metrics can indicate missing tokens. But does one care, that 20 of 100 requests are coming without token and one can see it in the logs?

I'd like to use the library as is and in case the client making the requests does not meet the pre-requisites of the call, he is informed about it, and that is the case given proper integration of this library here. So IMHO there's no need to log a WARN.